batch/batches/Cleanup/KAsyncDirectoryCleanup.class.php

```diff
@@ -25,64 +27,161 @@ public static function getType()
 	*/
 	public function run($jobs = null)
 	{
-		$path = $this->getAdditionalParams("path");
-		$pattern = $this->getAdditionalParams("pattern");
-		$simulateOnly = $this->getAdditionalParams("simulateOnly");
-		$minutesOld = $this->getAdditionalParams("minutesOld");
+		$path = $this->getAdditionalParams('path');
+		$pattern = $this->getAdditionalParams('pattern');
+		$simulateOnly = $this->getAdditionalParams('simulateOnly');
+		$minutesOld = $this->getAdditionalParams('minutesOld');
 		$searchPath = $path . $pattern;
-		KalturaLog::info("Searching [$searchPath]");
+		KalturaLog::info('Searching ' . $searchPath);
+		$usePHP = $this->getAdditionalParams('usePHP');
+		$this->deleteFiles($searchPath, $minutesOld, $simulateOnly, $usePHP);
+	}
+
+	/**
+	 * @param $searchPath
+	 * @param $minutesOld
+	 * @param $simulateOnly
+	 * @return bool
+	 */
+	protected function deleteFiles($searchPath, $minutesOld, $simulateOnly, $usePHP)
+	{
+		$secondsOld = $minutesOld * 60;
+		$files = kFile::getFilesByPattern($searchPath);
+		KalturaLog::info('Found [' . count($files) . '] to scan');
 
-		if($this->getAdditionalParams("usePHP"))
+		$now = time();
+		KalturaLog::info('Deleting files that are ' . $secondsOld . ' seconds old (modified before ' . date('c', $now - $secondsOld) . ')');
+		foreach ($files as $file)
 		{
-			$this->deleteFilesPHP($searchPath, $minutesOld, $simulateOnly);
+			$filemtime = kFile::getFileLastUpdatedTime($file);
+			if ($filemtime > $now - $secondsOld)
+			{
+				continue;
+			}
+
+			if ($simulateOnly)
+			{
+				KalturaLog::info('Simulating: Deleting file [' . $file . ' ], it\'s last modification time was ' . date('c', $filemtime));
+				continue;
+			}
+
+			if (kFile::checkIsDir($file))
+			{
+				if ($this->shouldDeleteDirectory($file, $now, $secondsOld))
+				{
+					$this->deleteDirectory($file, $usePHP);
+				}
+				else
+				{
+					continue;
+				}
+			}
+			else
+			{
+				$this->deleteFile($file, $usePHP);
+			}
 		}
-		else
+	}
+
+	/**
+	 * @param $path
+	 * @param $now
+	 * @param $secondsOld
+	 * @return bool
+	 */
+	protected function shouldDeleteDirectory($path, $now, $secondsOld)
```

Comment:
Logic problem.
If dir is not chunk , this function returns always true.
The logic should be - always not, if it is chunk - check internal content and decide yes of no.